godot-core/src/obj/script.rs

```diff
@@ -337,6 +342,37 @@ pub unsafe fn create_script_instance<T: ScriptInstance>(
     }
 }
 
+/// Checks if an instance of the script exists for a given object.
+///
+/// Use this function to implement [`IScriptExtension::instance_has`](crate::classes::IScriptExtension::instance_has).
+#[cfg(since_api = "4.2")]
+pub fn script_instance_exists<O, S>(object: &Gd<O>, script: &Gd<S>) -> bool
+where
+    O: Inherits<Object>,
+    S: Inherits<Script> + IScriptExtension + super::Bounds<Declarer = super::bounds::DeclUser>,
+{
+    let object_script_variant = object.upcast_ref().get_script();
+
+    if object_script_variant.is_nil() {
+        return false;
+    }
+
+    let object_script: Gd<Script> = object_script_variant.to();
+
+    if object_script.upcast_ref::<Script>().__object_ptr() != script.upcast_ref().__object_ptr() {
+        return false;
+    }
+
+    let Some(language) = script.bind().get_language() else {
+        return false;
+    };
+
+    let get_instance_fn = sys::interface_fn!(object_get_script_instance);
+    let instance = unsafe { get_instance_fn(object.obj_sys(), language.obj_sys()) };
```

Comment:
Is there any special consideration regarding when `object_get_script_instance` is safe to call?

If not, maybe simply add something like
```rs
// SAFETY: object and language are alive and their sys pointers are valid.
```

Reply:
No that sums it up. `object_get_script_instance` is quite simple.